File: live/src/main/java/com/hutcwp/live/livebiz/ui/component/publicmessage/psg/holder/ActivityNewsHolder.java
package com.hutcwp.live.livebiz.ui.component.publicmessage.psg.holder;

import android.graphics.drawable.Drawable;
import android.view.View;
import android.widget.TextView;

import com.hutcwp.live.livebiz.ui.component.publicmessage.lib.BaseChatViewHolder;
import com.hutcwp.live.livebiz.ui.component.publicmessage.psg.MyChatMsg;
import com.hutcwp.livebiz.R;

import androidx.core.content.ContextCompat;
import me.hutcwp.BasicConfig;
import me.hutcwp.util.ResolutionUtils;


/**
 * @author RyanLee
 */
public class ActivityNewsHolder extends BaseChatViewHolder {

    public ActivityNewsHolder(View itemView) {
        super(itemView);
    }

    @Override
    public void bindData(Object obj, int position) {
        MyChatMsg data = (MyChatMsg) obj;
        TextView textView = (TextView) getView(R.id.tv_activity_news);
        Drawable drawableLeft = ContextCompat.getDrawable(textView.getContext(), R.drawable.ic_activity);
        Drawable drawableRight = ContextCompat.getDrawable(textView.getContext(), R.drawable.ic_arrow);

        int int18 = (int) ResolutionUtils.convertDpToPixel(18f, BasicConfig.getApplicationContext());
        int dp12 = (int) ResolutionUtils.convertDpToPixel(12f, BasicConfig.getApplicationContext());
        if (drawableLeft != null) {
            drawableLeft.setBounds(0, 0, int18, int18);

        }
        if (drawableRight != null) {
            drawableRight.setBounds(0, 0, dp12, dp12);
        }
        textView.setCompoundDrawables(drawableLeft, null, drawableRight, null);
        textView.setText(data.activityNews);
    }
}

File: live/src/main/java/com/hutcwp/live/chat/binder/chat/ActivityNewsHolder.kt
package com.hutcwp.live.chat.binder.chat

import android.graphics.drawable.Drawable
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.drakeet.multitype.ItemViewBinder
import com.hutcwp.live.R
import com.hutcwp.live.chat.bean.MyChatMsg
import com.ryan.baselib.util.AppUtils
import com.ryan.baselib.util.DensityUtils

/**
 * @author RyanLee
 */
class ActivityNewsHolder: ItemViewBinder<MyChatMsg, ActivityNewsHolder.ViewHolder>() {

    override fun onCreateViewHolder(inflater: LayoutInflater, parent: ViewGroup): ViewHolder {
        return ViewHolder(inflater.inflate(R.layout.live_layout_activity_news, parent, false))
    }

    override fun onBindViewHolder(holder: ViewHolder, item: MyChatMsg) {
        val drawableLeft: Drawable? = ContextCompat.getDrawable(AppUtils.getContext(), R.drawable.ic_activity)
        val drawableRight: Drawable? = ContextCompat.getDrawable(AppUtils.getContext(), R.drawable.ic_arrow)
        drawableLeft?.setBounds(0, 0, DensityUtils.dp2px(AppUtils.getContext(), 18f), DensityUtils.dp2px(AppUtils.getContext(), 18f))
        drawableRight?.setBounds(0, 0, DensityUtils.dp2px(AppUtils.getContext(), 12f), DensityUtils.dp2px(AppUtils.getContext(), 12f))
        holder.textView.setCompoundDrawables(drawableLeft, null, drawableRight, null)
        holder.textView.text = item.activityNews

    }


    class ViewHolder(itemView : View): RecyclerView.ViewHolder(itemView) {
        val textView: TextView = itemView.findViewById(R.id.tv_activity_news)
    }
}